### DISCUSSION TITLE
Is it possible to get Transition's afterLeave behaviour when using transition props?

### DISCUSSION THREAD

**Discussion by jahvi** [tailwindlabs/headlessui#3587] [Help]
I'm using the `transition` prop in a `Dialog` component to animate when it's open or closed, however I want to reset some state after the dialog is fully closed.

I used to be able to do this using the `afterLeave` callback from the `Transition` component but I can't find a way to do the equivalent using props. Right now my workaround is to reset my state using a timeout but it's not ideal.

Is there a way to do this?

**Comment by HusamElbashir** [comment]
Same question. I'm currently resorting to using an `onTransitionEnd` event listener but it's not ideal since I have to filter events bubbling from child components that fire the same event.
